openapi_python_client/templates/model.pyi

```diff
@@ -105,27 +106,69 @@ class {{ model.reference.class_name }}:
             {{ construct(model.additional_properties, "prop_dict") | indent(12) }}
             additional_properties[prop_name] = {{ model.additional_properties.python_name }}
 
-        {{model.reference.module_name}}.additional_properties = additional_properties
+        {{model.reference.module_name}}._additional_properties = additional_properties
     {% else %}
-        {{model.reference.module_name}}.additional_properties = d
+        {{model.reference.module_name}}._additional_properties = d
     {% endif %}
 {% endif %}
         return {{model.reference.module_name}}
 
     {% if model.additional_properties %}
     @property
     def additional_keys(self) -> List[str]:
-        return list(self.additional_properties.keys())
+        return list(self._additional_properties.keys())
 
     def __getitem__(self, key: str) -> {{ additional_property_type }}:
-        return self.additional_properties[key]
+        return self._additional_properties[key]
 
     def __setitem__(self, key: str, value: {{ additional_property_type }}) -> None:
-        self.additional_properties[key] = value
+        self._additional_properties[key] = value
 
     def __delitem__(self, key: str) -> None:
-        del self.additional_properties[key]
+        del self._additional_properties[key]
 
     def __contains__(self, key: str) -> bool:
-        return key in self.additional_properties
+        return key in self._additional_properties
     {% endif %}
+
+{% macro required_property_type(prop) %}
+{%- if prop.nullable -%}
+Optional[{{prop.get_base_type_string()}}]
+{%- else -%}
+{{prop.get_base_type_string()}}
+{%- endif -%}
+{% endmacro %}
+
+    {% if model.additional_properties %}
```

Comment:
This really deserves to be upstreamed. Do you mind submitting a PR against `openapi-python-client`? If it doesn't get accepted in time, we'll move forward with this, but worth a shot I think.

Reply:
Yup.  Gives us the opportunity to upstream the required property type onto the property as well.